src/vs/workbench/parts/files/browser/fileActions.ts

```diff
@@ -1990,6 +1993,58 @@ export function getWellFormedFileName(filename: string): string {
 	return filename;
 }
 
+export const SHOW_MODIFICATIONS_SCHEME = 'showModifications';
+export class ShowModificationsAction extends Action implements ITextModelContentProvider {
```

Comment:
I suggest to also make this action available via keybinding and command list. As such it should be registered as a global action and in that case just use the current active editor as thing to compare. The action would simply do nothing if there is no active file opened or that file is not dirty.